scripts/gen_help_html.lua

```diff
@@ -983,7 +987,6 @@ local function gen_one(fname, text, to_fname, old, commit, parser_path)
     n = n + 1 + #h1.subheadings
   end
   for _, h1 in ipairs(headings) do
-    ---@type string
     toc = toc .. ('<div class="help-toc-h1"><a href="#%s">%s</a>\n'):format(h1.tag, h1.name)
```

Comment:
Just to note, this is one area where LuaLS can be flaky as `toc` is on both sides of the assignment.

Reply:
Fixed it by just using `string.format` instead.